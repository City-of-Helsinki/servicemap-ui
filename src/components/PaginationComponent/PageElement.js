import React from 'react';
import PropTypes from 'prop-types';
import { intlShape } from 'react-intl';
import { Typography, withStyles, ButtonBase } from '@material-ui/core';
import styles from './styles';

// Page number element
const PageElement = ({
  classes, className, intl, isActive, number, onClick, ...rest
}) => {
  const newClassName = `${className} ${classes.pageItem} ${isActive ? classes.pageElementActive : classes.pageElement}`;
  return (
    <li>
      <ButtonBase
        role="link"
        disabled={isActive}
        onClick={onClick}
        tabIndex={isActive ? null : '0'}
      >
        <Typography
          variant="subtitle1"
          component="p"
          className={newClassName}
          {...rest}
        >
          <Typography variant="srOnly">
            {
            isActive
              ? intl.formatMessage({ id: 'general.pagination.currentlyOpenedPage' }, { count: number })
              : intl.formatMessage({ id: 'general.pagination.openPage' }, { count: number })
          }
          </Typography>
          <span aria-hidden="true">
            {number}
          </span>
        </Typography>
      </ButtonBase>
    </li>
  );
};

PageElement.propTypes = {
  classes: PropTypes.objectOf(PropTypes.any).isRequired,
  className: PropTypes.string,
  intl: intlShape.isRequired,
  isActive: PropTypes.bool.isRequired,
  number: PropTypes.number.isRequired,
  onClick: PropTypes.func.isRequired,
};

PageElement.defaultProps = {
  className: '',
};

export default withStyles(styles)(PageElement);
