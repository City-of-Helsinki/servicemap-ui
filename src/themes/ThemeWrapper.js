import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { ThemeProvider } from '@mui/material/styles';
import themes from '.';

const mapStateToProps = (state) => {
  const { user } = state;
  return {
    theme: user.theme,
  };
};

// Component to handle theme changes
const ThemeHandler = ({ children, theme }) => (
// Get correct theme setting from store
  <ThemeProvider theme={theme === 'dark' ? themes.SMThemeDark : themes.SMTheme}>
    {children}
  </ThemeProvider>
);
ThemeHandler.propTypes = {
  children: PropTypes.node,
  theme: PropTypes.string.isRequired,
};
ThemeHandler.defaultProps = {
  children: null,
};

const ThemeWrapper = connect(
  mapStateToProps,
)(ThemeHandler);

export default ThemeWrapper;
