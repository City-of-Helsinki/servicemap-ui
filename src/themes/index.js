import { createMuiTheme } from '@material-ui/core';

const focusIndicator = {
  outline: '2px solid transparent',
  boxShadow: '0 0 0 2px rgb(255, 255, 255), 0 0 0 6px rgb(71, 131, 235), 0 0 0 8px rgb(255, 255, 255)',
};

const focusIndicatorDark = {
  outline: '2px solid transparent',
  boxShadow: '0 0 0 2px rgb(255, 255, 255), 0 0 0 6px rgb(0, 0, 0), 0 0 0 8px rgb(255, 255, 255)',
};

const props = {
  // Globally disable all default mui focus effects
  MuiButtonBase: {
    disableRipple: true,
  },
  MuiButton: {
    disableRipple: true,
    disableFocusRipple: true,
  },
  MuiTypography: {
    variant: 'body2',
  },
};

const overrides = theme => ({
  MuiButtonBase: {
    root: {
      // Default keyboard focus indicator for buttons
      '&$focusVisible': theme === 'dark' ? focusIndicatorDark : focusIndicator,
    },
  },
  MuiInputBase: {
    // Default keyboard focus indicator for input fields
    root: { '&$focused': theme === 'dark' ? focusIndicatorDark : focusIndicator },
  },
  MuiListItemText: {
    root: {
      marginTop: 0,
      marginBottom: 0,
    },
  },
  MuiSvgIcon: {
    root: {
      fontSize: 24,
    },
  },
  PrivateSwitchBase: {
    root: {
      padding: 12,
    },
  },
});

const breakpoints = {
  values: {
    xs: 0,
    sm: 700,
    md: 900,
    lg: 1280,
    xl: 1920,
  },
};

const spacing = 8;

const custom = {
  body2light: {
    color: 'rgba(0,0,0,0.7)',
  },
};

const zIndex = {
  behind: -1,
  forward: 50,
  sticky: 51,
  infront: 900,
};

const typography = {
  useNextVariants: true,
  fontSize: 16,
  // Use the system font instead of the default Roboto font.
  fontFamily: [
    'Lato',
  ].join(','),
  body1: {
    fontSize: '1.043rem',
    letterSpacing: '0.03125rem', // 0.5px
    lineHeight: '1.75rem',
  },
  body2: {
    fontSize: '0.913rem',
    letterSpacing: '0.015625rem', // 0.25px
    lineHeight: '1.25rem',
  },
  subtitle1: {
    fontSize: '1.03rem',
    fontWeight: 'bold',
    letterSpacing: '0.00937rem', // 0.15px
    lineHeight: '1.5rem',
  },
  caption: {
    fontSize: '0.7725rem', // 12.36px
    fontWeight: 'bold',
    letterSpacing: '0.025rem', // 0.4px
    lineHeight: '1rem',
    color: 'rgba(0,0,0,0.6)',
  },
  h1: {
    fontSize: '6.259rem',
    letterSpacing: '-0.09375em', // -1.5px
  },
  h2: {
    fontSize: '3.862rem',
    fontWeight: 'bold',
    letterSpacing: '-0.03125rem', // -0.5px
  },
  h3: {
    fontSize: '3.089rem',
    fontWeight: 'bold',
    letterSpacing: 0,
  },
  h4: {
    fontSize: '2.214rem',
    letterSpacing: '0.015625rem', // 0.25px
  },
  h5: {
    fontSize: '1.564rem',
    letterSpacing: 0,
  },
  h6: {
    fontSize: '1.288rem',
    fontWeight: 'bold',
    letterSpacing: '0.015625rem', // 0.25px
    lineHeight: '1.5rem',
  },
  button: {
    fontSize: '0.90125rem', // 14.42px
    fontWeight: 'bold',
    letterSpacing: '0.078125rem', // 1.25px
    lineHeight: '1rem', // 16px
    color: '#ffffff',
  },
  overline: {
    fontSize: '0.7725rem', // 12.36px
    fontWeight: 'bold',
    letterSpacing: '0.125rem', // 2px
    lineHeight: '1rem',
  },
};

// Color palette for normal theme
export const paletteDefault = {
  primary: {
    main: 'rgb(25, 100, 230)',
    highContrast: '#fff',
  },
  secondary: {
    main: 'rgb(43, 47, 57)',
    hover: '#1d39ad',
  },
  background: {
    main: 'linear-gradient(340.58deg, rgb(11, 123, 237) 0%, rgb(20, 108, 232) 67.04%, rgb(25, 100, 230) 100%)',
    plain: 'rgb(20, 108, 232)',
    front: 'linear-gradient(340.58deg, rgb(11, 123, 237) 0%, rgb(20, 108, 232) 67.04%, rgb(25, 100, 230) 100%)',
  },
  white: {
    light: '#f2f2f2',
    main: '#FFFFFF',
    dark: '#949494',
    contrastText: '#000',
  },
  detail: {
    main: 'rgb(25, 100, 230)',
    alpha: 'rgba(25,100,230,0.5)',
  },
  disabled: {
    main: 'rgb(239, 239, 239)',
    strong: '#898989',
  },
  focusBorder: {
    main: 'rgb(71, 131, 235)',
  },
  warning: {
    main: '#AF0021',
  },
};

// Color palette for dark theme
export const paletteDark = {
  primary: {
    main: '#353638',
    highContrast: '#fff',
  },
  secondary: {
    main: 'rgb(43, 47, 57)',
    hover: '#5b5d61',
  },
  background: {
    main: '#4A4A4C',
    plain: '#4A4A4C',
    front: 'linear-gradient(326.21deg, rgba(0, 0, 0, 0.79) 0%, rgba(71, 71, 71, 0.79) 100%)',
  },
  white: {
    light: '#f2f2f2',
    main: '#FFFFFF',
    dark: '#949494',
    contrastText: '#000',
  },
  detail: {
    main: '#585B63',
    alpha: 'rgba(88,91,99,0.5)',
  },
  disabled: {
    main: 'rgb(239, 239, 239)',
    strong: '#898989',
  },
  warning: {
    main: '#AF0021',
  },
  focusBorder: {
    main: '#000',
  },
};

// Themes
const SMTheme = createMuiTheme({
  props,
  overrides: overrides('default'),
  breakpoints,
  typography,
  spacing,
  custom,
  palette: paletteDefault,
  zIndex,
});

const SMThemeDark = createMuiTheme({
  props,
  overrides: overrides('dark'),
  breakpoints,
  typography,
  spacing,
  custom,
  palette: paletteDark,
  zIndex,
});


export default { SMTheme, SMThemeDark };
