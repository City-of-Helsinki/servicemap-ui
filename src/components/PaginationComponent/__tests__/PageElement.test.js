// Link.react.test.js
import React from 'react';
import { createMount } from '@material-ui/core/test-utils';
import { MuiThemeProvider } from '@material-ui/core';
import { IntlProvider } from 'react-intl';
import themes from '../../../themes';
import PageElement from '../PageElement';

// Mock props for intl provider
const intlMock = {
  locale: 'en',
  messages: {
    'general.pagination.openPage': 'Open page {count}',
    'general.pagination.currentlyOpenedPage': 'Sivu {count}, avattu',
  },
  wrapRichTextChunksInFragment: false,
};

// Generic required props for ResultItem
const mockProps = {
  number: 2,
  onClick: jest.fn(),
  isActive: false,
};

// eslint-disable-next-line react/prop-types
const Providers = ({ children }) => (
  <IntlProvider {...intlMock}>
    <MuiThemeProvider theme={themes.SMTheme}>
      {children}
    </MuiThemeProvider>
  </IntlProvider>
);

describe('<PageElement />', () => {
  // let render;
  let mount;

  beforeEach(() => {
    // render = createRender({ wrappingComponent: Providers });
    mount = createMount({ wrappingComponent: Providers });
  });

  afterEach(() => {
    mount.cleanUp();
  });

  it('should work', () => {
    const component = mount(<PageElement {...mockProps} />);
    expect(component).toMatchSnapshot();
  });

  it('simulates click event', () => {
    const mockCallBack = jest.fn();
    const component = mount(<PageElement {...mockProps} onClick={mockCallBack} />);

    component.find('ForwardRef(ButtonBase)').simulate('click');
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });

  it('simulates keyboard event', () => {
    const mockCallBack = jest.fn();
    const component = mount(<PageElement {...mockProps} onClick={mockCallBack} />);

    component.find('ForwardRef(ButtonBase)').simulate('keyDown', { which: 13 });
    component.find('ForwardRef(ButtonBase)').simulate('keyDown', { which: 32 });
    expect(mockCallBack.mock.calls.length).toEqual(2);
  });

  it('does show text correctly', () => {
    const component = mount(<PageElement {...mockProps} />);

    const p = component.find('p span');
    expect(p.at(1).text()).toEqual(`${mockProps.number}`);
  });

  it('does set active correctly', () => {
    const component = mount(<PageElement {...mockProps} isActive />);
    expect(component.find('ForwardRef(ButtonBase)').props().disabled).toBeTruthy();
  });

  it('does use default accessibility attributes correctly', () => {
    const component = mount(<PageElement {...mockProps} />);

    // Expect screen reader texts to render correctly
    const srText = component.find('p span').at(0).text();
    expect(srText).toEqual('Open page 2');

    // // Expect aria-hidden attributes to be placed correctly
    const spans = component.find('p span');
    expect(spans.at(0).props()['aria-hidden']).toBeFalsy();
    expect(spans.at(1).props()['aria-hidden']).toBeTruthy();

    // // Expect role to be set to link
    expect(component.find('ForwardRef(ButtonBase)').props().role).toEqual('link');

    // // Expect element to have tabIndex 0
    expect(component.find('ForwardRef(ButtonBase)').props().tabIndex).toEqual('0');
  });

  it('does use given accessibility attributes correctly', () => {
    const component = mount(
      <PageElement
        {...mockProps}
        isActive
      />,
    );

    const srText = component.find('p span').at(0).text();
    expect(srText).toEqual(`Sivu ${mockProps.number}, avattu`);

    // // Expect element to have tabIndex -1
    expect(component.find('ForwardRef(ButtonBase)').props().tabIndex).toEqual('-1');
  });
});
