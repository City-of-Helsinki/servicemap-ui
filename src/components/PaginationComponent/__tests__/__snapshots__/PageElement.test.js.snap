// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageElement /> should work 1`] = `
<injectIntl(WithStyles(PageElement))
  isActive={false}
  number={2}
  onClick={[MockFunction]}
>
  <WithStyles(PageElement)
    intl={
      Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatDateToParts": [Function],
        "formatDisplayName": [Function],
        "formatList": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatNumberToParts": [Function],
        "formatPlural": [Function],
        "formatRelativeTime": [Function],
        "formatTime": [Function],
        "formatTimeToParts": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getDisplayNames": [Function],
          "getListFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralRules": [Function],
          "getRelativeTimeFormat": [Function],
        },
        "locale": "en",
        "messages": Object {
          "general.pagination.currentlyOpenedPage": "Sivu {count}, avattu",
          "general.pagination.openPage": "Open page {count}",
        },
        "onError": [Function],
        "textComponent": Symbol(react.fragment),
        "timeZone": undefined,
        "wrapRichTextChunksInFragment": false,
      }
    }
    isActive={false}
    number={2}
    onClick={[MockFunction]}
  >
    <PageElement
      className=""
      classes={
        Object {
          "arrowFlip": "PageElement-arrowFlip-2",
          "arrowIcon": "PageElement-arrowIcon-1",
          "borderBlack": "PageElement-borderBlack-3",
          "button": "PageElement-button-4",
          "buttonContainer": "PageElement-buttonContainer-5",
          "list": "PageElement-list-6",
          "listContainer": "PageElement-listContainer-7",
          "pageElement": "PageElement-pageElement-8",
          "pageElementActive": "PageElement-pageElementActive-9",
          "pageItem": "PageElement-pageItem-10",
        }
      }
      intl={
        Object {
          "defaultFormats": Object {},
          "defaultLocale": "en",
          "formatDate": [Function],
          "formatDateToParts": [Function],
          "formatDisplayName": [Function],
          "formatList": [Function],
          "formatMessage": [Function],
          "formatNumber": [Function],
          "formatNumberToParts": [Function],
          "formatPlural": [Function],
          "formatRelativeTime": [Function],
          "formatTime": [Function],
          "formatTimeToParts": [Function],
          "formats": Object {},
          "formatters": Object {
            "getDateTimeFormat": [Function],
            "getDisplayNames": [Function],
            "getListFormat": [Function],
            "getMessageFormat": [Function],
            "getNumberFormat": [Function],
            "getPluralRules": [Function],
            "getRelativeTimeFormat": [Function],
          },
          "locale": "en",
          "messages": Object {
            "general.pagination.currentlyOpenedPage": "Sivu {count}, avattu",
            "general.pagination.openPage": "Open page {count}",
          },
          "onError": [Function],
          "textComponent": Symbol(react.fragment),
          "timeZone": undefined,
          "wrapRichTextChunksInFragment": false,
        }
      }
      isActive={false}
      number={2}
      onClick={[MockFunction]}
    >
      <li>
        <WithStyles(ForwardRef(ButtonBase))
          disabled={false}
          onClick={[MockFunction]}
          onKeyDown={[Function]}
          role="link"
          tabIndex="0"
        >
          <ForwardRef(ButtonBase)
            classes={
              Object {
                "disabled": "Mui-disabled",
                "focusVisible": "Mui-focusVisible",
                "root": "MuiButtonBase-root",
              }
            }
            disableRipple={true}
            disabled={false}
            onClick={[MockFunction]}
            onKeyDown={[Function]}
            role="link"
            tabIndex="0"
          >
            <button
              className="MuiButtonBase-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[MockFunction]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="link"
              tabIndex="0"
              type="button"
            >
              <WithStyles(ForwardRef(Typography))
                className=" PageElement-pageItem-10 PageElement-pageElement-8"
                component="p"
                variant="subtitle1"
              >
                <ForwardRef(Typography)
                  className=" PageElement-pageItem-10 PageElement-pageElement-8"
                  classes={
                    Object {
                      "alignCenter": "MuiTypography-alignCenter",
                      "alignJustify": "MuiTypography-alignJustify",
                      "alignLeft": "MuiTypography-alignLeft",
                      "alignRight": "MuiTypography-alignRight",
                      "body1": "MuiTypography-body1",
                      "body2": "MuiTypography-body2",
                      "button": "MuiTypography-button",
                      "caption": "MuiTypography-caption",
                      "colorError": "MuiTypography-colorError",
                      "colorInherit": "MuiTypography-colorInherit",
                      "colorPrimary": "MuiTypography-colorPrimary",
                      "colorSecondary": "MuiTypography-colorSecondary",
                      "colorTextPrimary": "MuiTypography-colorTextPrimary",
                      "colorTextSecondary": "MuiTypography-colorTextSecondary",
                      "displayBlock": "MuiTypography-displayBlock",
                      "displayInline": "MuiTypography-displayInline",
                      "gutterBottom": "MuiTypography-gutterBottom",
                      "h1": "MuiTypography-h1",
                      "h2": "MuiTypography-h2",
                      "h3": "MuiTypography-h3",
                      "h4": "MuiTypography-h4",
                      "h5": "MuiTypography-h5",
                      "h6": "MuiTypography-h6",
                      "noWrap": "MuiTypography-noWrap",
                      "overline": "MuiTypography-overline",
                      "paragraph": "MuiTypography-paragraph",
                      "root": "MuiTypography-root",
                      "srOnly": "MuiTypography-srOnly",
                      "subtitle1": "MuiTypography-subtitle1",
                      "subtitle2": "MuiTypography-subtitle2",
                    }
                  }
                  component="p"
                  variant="subtitle1"
                >
                  <p
                    className="MuiTypography-root  PageElement-pageItem-10 PageElement-pageElement-8 MuiTypography-subtitle1"
                  >
                    <WithStyles(ForwardRef(Typography))
                      variant="srOnly"
                    >
                      <ForwardRef(Typography)
                        classes={
                          Object {
                            "alignCenter": "MuiTypography-alignCenter",
                            "alignJustify": "MuiTypography-alignJustify",
                            "alignLeft": "MuiTypography-alignLeft",
                            "alignRight": "MuiTypography-alignRight",
                            "body1": "MuiTypography-body1",
                            "body2": "MuiTypography-body2",
                            "button": "MuiTypography-button",
                            "caption": "MuiTypography-caption",
                            "colorError": "MuiTypography-colorError",
                            "colorInherit": "MuiTypography-colorInherit",
                            "colorPrimary": "MuiTypography-colorPrimary",
                            "colorSecondary": "MuiTypography-colorSecondary",
                            "colorTextPrimary": "MuiTypography-colorTextPrimary",
                            "colorTextSecondary": "MuiTypography-colorTextSecondary",
                            "displayBlock": "MuiTypography-displayBlock",
                            "displayInline": "MuiTypography-displayInline",
                            "gutterBottom": "MuiTypography-gutterBottom",
                            "h1": "MuiTypography-h1",
                            "h2": "MuiTypography-h2",
                            "h3": "MuiTypography-h3",
                            "h4": "MuiTypography-h4",
                            "h5": "MuiTypography-h5",
                            "h6": "MuiTypography-h6",
                            "noWrap": "MuiTypography-noWrap",
                            "overline": "MuiTypography-overline",
                            "paragraph": "MuiTypography-paragraph",
                            "root": "MuiTypography-root",
                            "srOnly": "MuiTypography-srOnly",
                            "subtitle1": "MuiTypography-subtitle1",
                            "subtitle2": "MuiTypography-subtitle2",
                          }
                        }
                        variant="srOnly"
                      >
                        <span
                          className="MuiTypography-root MuiTypography-srOnly"
                        >
                          Open page 2
                        </span>
                      </ForwardRef(Typography)>
                    </WithStyles(ForwardRef(Typography))>
                    <span
                      aria-hidden="true"
                    >
                      2
                    </span>
                  </p>
                </ForwardRef(Typography)>
              </WithStyles(ForwardRef(Typography))>
            </button>
          </ForwardRef(ButtonBase)>
        </WithStyles(ForwardRef(ButtonBase))>
      </li>
    </PageElement>
  </WithStyles(PageElement)>
</injectIntl(WithStyles(PageElement))>
`;
